spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
One thing about the current `modules` property is ordering - which for becomes very important for direct binding. 
How about we do this: 
`MODULES=[foo]coordinates1,[bar]coordinates2,[baz]coordinates3` with labeling being optional and defaulted to artifact/position (e.g `time-source_0`)
And then arguments can come in the form 
`FOO_FOO=bar` or `TIME_SOURCE_0`.